src/DbPostgreSQLAdapter/FieldValue.cpp

```diff
@@ -0,0 +1,495 @@
+#include "stdafx.h"
+#include "FieldValue.h"
+
+namespace{
+	bool isDateTimeNull(const std::chrono::system_clock::time_point& dateTime)
+	{
+		return dateTime == std::chrono::system_clock::time_point();
+	}
+
+	std::chrono::system_clock::time_point stringISOToDateTime(const std::string& dateTime)
+	{
+		std::chrono::system_clock::time_point timePointDateTime;
+		std::istringstream{ dateTime } >> std::chrono::parse("%F %T%z", timePointDateTime);
+		return timePointDateTime;
+	}
+}
+
+namespace systelab::db::postgresql {
+
+	FieldValue::FieldValue(const IField& field)
+		: m_field(field)
+		, m_nullValue(true)
+		, m_default(false)
+		, m_boolValue(false)
+		, m_intValue(0)
+		, m_doubleValue(0.)
+		, m_stringValue("")
+		, m_dateTimeValue()
+		, m_binaryValue()
+	{
+	}
+
+	FieldValue::FieldValue(const IField& field, bool value)
+		: m_field(field)
```

Comment:
Since you are relying on copy-constructor, all the members are already initialized there and you should just overwrite the `m_nullValue`.

Reply:
Actually this is not a copy constructor, IField is an abstraction of the FieldValue type. What I tried was to use the first constructor to preinitialize everything, and then overwrite the specific parameters of each constructor, but it didn't worked out. It seems that the compiler thas not like to overwrite members that are already initialized by another nested constructor.